src/API/NavigationFavorites.php

```diff
@@ -0,0 +1,299 @@
+					'args'                => array(
+						'item_id' => array(
+							'required' => true,
+						),
+						'user_id' => array(
+							'required'          => true,
+							'validate_callback' => function( $param, $request, $key ) {
+								return is_numeric( $param );
+							},
+						),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => array(
+						'item_id' => array(
+							'required' => true,
+						),
+						'user_id' => array(
+							'required'          => true,
+							'validate_callback' => function( $param, $request, $key ) {
+								return is_numeric( $param );
+							},
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/me',
+			array(
+				array(
+					'methods'  => \WP_REST_Server::READABLE,
+					'callback' => array( $this, 'get_items_by_current_user' ),
+				),
+				array(
+					'methods'  => \WP_REST_Server::CREATABLE,
+					'callback' => array( $this, 'add_item_by_current_user' ),
+					'args'     => array(
+						'item_id' => array(
+							'required' => true,
+						),
+					),
+				),
+				array(
+					'methods'  => \WP_REST_Server::DELETABLE,
+					'callback' => array( $this, 'delete_item_by_current_user' ),
+					'args'     => array(
+						'item_id' => array(
+							'required' => true,
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+	}
+
+	/**
+	 * Get all favorites.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$user_id = $request->get_param( 'user_id' );
+
+		$response = Favorites::get_all( $user_id );
+
+		if ( is_wp_error( $response ) || ! $response ) {
+			return rest_ensure_response( $this->prepare_error( $response ) );
+		}
+
+		return rest_ensure_response(
+			array_map( 'stripslashes', $response )
+		);
+	}
+
+	/**
+	 * Get all favorites of current user.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function get_items_by_current_user( $request ) {
+		$current_user = get_current_user_id();
+
+		if ( ! $current_user ) {
+			return new \WP_Error(
+				'woocommerce_favorites_invalid_user',
+				__( 'You must be authenticated to use this endpoint', 'woocommerce-admin' ),
+				array( 'status' => 401 )
+			);
+		}
+
+		$request->set_param( 'user_id', $current_user );
+
+		return $this->get_items( $request );
+	}
+
+	/**
+	 * Add a favorite.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function add_item( $request ) {
+		$user_id = $request->get_param( 'user_id' );
```

Comment:
Might be nice to check if this user (ID) exists before adding data.  Currently responds with an empty array for getting from a fake user ID and will allow me to add to random user IDs.

I'm okay tackling this in a follow-up too.